spec/unit/puppet_x/puppetlabs/scheduled_task/task_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'spec_helper'
+require 'puppet_x/puppetlabs/scheduled_task/task'
+
+describe PuppetX::PuppetLabs::ScheduledTask::Task do
+  describe "with account with password" do
+
+    before(:each) do
+      @password = "blah_#{rand(999)}"
+      principal = double('principal')
+      principal.stub(:UserId).and_return('vagrant')
+      principal.stub(:UserId=).with(anything)
+      principal.stub(:LogonType=).with(anything)
+      principal.stub(:RunLevel=).with(anything)
+
+      definition = double('definition')
+      definition.stub(:Principal).and_return(principal)
+
+      _task = double('_task')
+      _task.stub(:Definition).and_return(definition)
+
+      folder = double('folder')
+      folder.stub(:GetTask).and_return(_task)
+
+      service = double('service')
+      service.stub(:GetFolder).and_return(folder)
+
+      allow(PuppetX::PuppetLabs::ScheduledTask::Task).to receive(:task_service).and_return(service)
+
+      @task = PuppetX::PuppetLabs::ScheduledTask::Task.new('test')
+    end
+
+    it "should set the password instance variable" do
```

Comment:
I'm not sure we'll regress on this behavior... this is testing internal class behavior and adds a lot of code for not a lot of extra risk aversion IMHO.